docs/SplunkOperatorUpgrade.md

```diff
@@ -0,0 +1,77 @@
+​
+* For general information about Splunk Enterprise compatibility and the upgrade process, see [How to upgrade Splunk Enterprise](https://docs.splunk.com/Documentation/Splunk/latest/Installation/HowtoupgradeSplunk).
+​
+* If you use forwarders, verify the Splunk Enterprise version compatibility with the forwarders in the [Compatibility between forwarders and Splunk Enterprise indexers](https://docs.splunk.com/Documentation/Forwarder/latest/Forwarder/Compatibilitybetweenforwardersandindexers) documentation.
+​
+## Upgrading Splunk Operator and Splunk Operator Deployment
+​
+1. Download the latest Splunk Operator installation yaml file.
+​
+```
+wget -O splunk-operator.yaml https://github.com/splunk/splunk-operator/releases/download/0.2.2/splunk-operator-install.yaml 
+```
+​
+2. (Optional) Review the file and update it with your specific customizations used during your install. 
+​
+3. Upgrade the Splunk Operator.
+​
+```
+kubectl apply -f splunk-operator.yaml
+```
+​
+After applying the yaml, a new operator pod will be created and the existing operator pod will drop. Example:
+​
+```
+kubectl get pods
+NAME                               READY   STATUS    RESTARTS   AGE
+splunk-operator-75f5d4d85b-8pshn   1/1     Running   0          5s
+​
+```
+​
+If a Splunk Operator release changes the custom resource (CRD) API version, the administrator is responsible for updating their Custom Resource specification to reference the latest CRD API version.
+​
+If a Splunk Operator release includes an updated Splunk Enterprise Docker image, the operator upgrade will also initiate pod restart using the latest Splunk Enterprise Docker image.
+​
+## Verifiy Upgrade is Successful
+​
+To verify the Splunk Operator has been upgraded to the release image in splunk-operator-install.yaml, you can check the version of the operator image. Example:
+​
+```
+kubectl get pods splunk-operator-6bcb8dd8f4-glkxt -o yaml | grep -i image
+image: docker.io/splunk/splunk-operator:0.2.2
+imagePullPolicy: IfNotPresent 
+```
+​
+To verify that a new Splunk Enterprise Docker image was applied to a pod, you can check the version of the image. Example:
+​
+```
+kubectl get pods splunk-default-monitoring-console-0 -o yaml | grep -i image
+image: splunk/splunk:8.1.2
+imagePullPolicy: IfNotPresent
+image: splunk/splunk:8.1.2
+```
+​
+## Splunk Enterprise Cluster upgrade example
+This is an example of the process followed by the Splunk Operator if the operator version is upgraded and a later Splunk Enterprise Docker image is available:
+​
+1. A new Splunk Operator pod will be created, and any existing operator pod will drop.
+2. If a License Master (LM) is present, the LM pod will be dropped. 
+3. A new LM pod is started with the updated specifications.
+4. After the new LM pod is up, any Cluster Manager (CM) and Deployer pods will be dropped.
```

Comment:
I think the order needs to be re-worked here:

1. A new Splunk Operator pod will be created, and the existing operator pod will be terminated.
2. Any existing License Master, Search Head, Deployer, ClusterMaster, Standalone pods will be terminated to be redeployed with the upgraded spec.
3. After a ClusterMaster pod is restarted, the Indexer Cluster pods which are connected to it are terminated and redeployed.
4. After all pods in the Indexer cluster and Search head cluster are redeployed, the Monitoring Console pod is terminated and redeployed.

(Note: If there are multiple pods per Custom Resource, the pods are terminated and re-deployed in a descending order with the highest numbered pod going first)

In a SHC scenario, the Search Head Cluster pods are not dependent on the deployer restart to complete. They act as separate entities. Only the indexer pods are dependent on the ClusterMaster they are connected to i.e they terminate and redeploy only once the ClusterMaster is up and running. @gaurav-splunk @sgontla @kashok-splunk, could you review and see if this is correct?

Reply:
You are correct about behavior with Deployer and SHC. Will update the example.